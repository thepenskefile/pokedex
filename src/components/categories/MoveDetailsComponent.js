// @flow

import React from 'react';
import { Container, Box, Flex, Heading, Columns, Column, Set, Tag, Text, palette } from 'fannypack';
import Counter from '../../ui/Counter';

type Props = {
  data: Object
};

const MoveDetailsComponent = ({ data }: Props) => (
  <Container textAlign="left" marginLeft="10px" width="96%">
    <Flex alignItems="center" marginBottom="10px">
      <Heading>{data.name.charAt(0).toUpperCase() + data.name.slice(1)}</Heading>
    </Flex>
    <Set marginBottom="20px" spacing="major-2" display="block">
      <Text use="strong">Contest type: </Text>
      {data.contest_type.name}
      <Text use="strong">Damage class: </Text>
      {data.damage_class.name}
    </Set>
    <Set marginBottom="20px">
      <Tag key={data.type.name} backgroundColor={data.type.name} size="medium">
        {data.type.name}
      </Tag>
    </Set>
    <Box marginTop="40px">
      <Columns marginTop="20px" marginBottom="20px" minBreakpoint="mobile">
        {data.accuracy && (
          <Column display="inline-block" spread={2} spreadTablet={4} spreadMobile={6} spreadWidescreen={3}>
            <Counter percentage={data.accuracy} label={'Accuracy'} color={palette(data.type.name)} />
          </Column>
        )}
        {data.pp && (
          <Column display="inline-block" spread={2} spreadTablet={4} spreadMobile={6} spreadWidescreen={3}>
            <Counter percentage={data.pp} label={'PP'} color={palette(data.type.name)} />
          </Column>
        )}
        {data.power && (
          <Column display="inline-block" spread={2} spreadTablet={4} spreadMobile={6} spreadWidescreen={3}>
            <Counter percentage={data.power} label={'Power'} color={palette(data.type.name)} />
          </Column>
        )}
      </Columns>
    </Box>
  </Container>
);

export default MoveDetailsComponent;
