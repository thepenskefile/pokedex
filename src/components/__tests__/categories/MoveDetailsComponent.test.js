import React from 'react';
import MoveDetailsComponent from '../../categories/MoveDetailsComponent';
import renderer from 'react-test-renderer';

const defaultProps = {
  data: {
    name: 'mockName',
    accuracy: 'mockAccuracy',
    pp: 'mockPP',
    power: 'mockPower',
    contest_type: { name: 'mockContestTypeName' },
    damage_class: { name: 'mockDamageClassName' },
    type: { name: 'mockTypeName' }
  }
};

it('renders correctly', () => {
  const component = renderer.create(<MoveDetailsComponent {...defaultProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
