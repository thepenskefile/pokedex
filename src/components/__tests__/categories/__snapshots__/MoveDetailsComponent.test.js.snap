// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="styled-sc-144wh3b-0 iUlbjp Box-sc-6sssmg-0 inpIDs"
  style={
    Object {
      "marginLeft": "10px",
      "textAlign": "left",
      "width": "96%",
    }
  }
  width="96%"
>
  <div
    className="Flex-sp5dv1-0 ddZvmG Box-sc-6sssmg-0 inpIDs"
    style={
      Object {
        "alignItems": "center",
        "marginBottom": "10px",
      }
    }
  >
    <h1
      className="heading styled-cya4kb-0 iFEcqF Heading-uhfcou-0 bnhnuY Box-sc-6sssmg-0 inpIDs"
    >
      MockName
    </h1>
  </div>
  <div
    className="Flex-sp5dv1-0 styled-sc-18btutf-0 kkaizR Box-sc-6sssmg-0 inpIDs"
    display="block"
    spacing="major-2"
    style={
      Object {
        "display": "block",
        "marginBottom": "20px",
      }
    }
  >
    <strong
      className="styled-sc-9qokh7-0 skDAJ Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Contest type: 
    </strong>
    mockContestTypeName
    <strong
      className="styled-sc-9qokh7-0 skDAJ Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Damage class: 
    </strong>
    mockDamageClassName
  </div>
  <div
    className="Flex-sp5dv1-0 styled-sc-18btutf-0 kkaizR Box-sc-6sssmg-0 inpIDs"
    spacing="minor-2"
    style={
      Object {
        "marginBottom": "20px",
      }
    }
  >
    <div
      className="Flex-sp5dv1-0 InlineFlex-sc-129poqw-0 styled__Tag-sc-1g01kou-2 gajwsF Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "backgroundColor": "mockTypeName",
        }
      }
    >
      <div
        className="styled__TagContent-sc-1g01kou-1 iuaGjz Box-sc-6sssmg-0 inpIDs"
      >
        mockTypeName
      </div>
    </div>
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
    style={
      Object {
        "marginTop": "40px",
      }
    }
  >
    <div
      className="styled-otaduv-0 wkSBA Box-sc-6sssmg-0 inpIDs"
      style={
        Object {
          "marginBottom": "20px",
          "marginTop": "20px",
        }
      }
    >
      <div
        className="styled__Column-sc-1lmgd0f-0 giZWgM Box-sc-6sssmg-0 inpIDs"
        display="inline-block"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <div
          className="Box-sc-6sssmg-0 inpIDs"
        >
          <svg
            className="sc-bdVaJa fJZkWb"
            viewBox="0 0 36 36"
          >
            <path
              className="sc-bwzfXH fUxNdF"
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <path
              className="sc-bxivhb dsVxfS"
              color={[Function]}
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <text
              className="sc-htpNat jJPiwm"
              x="18"
              y="20.35"
            >
              mockAccuracy
            </text>
          </svg>
          <div
            className="Box-sc-6sssmg-0 inpIDs"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Accuracy
          </div>
        </div>
      </div>
      <div
        className="styled__Column-sc-1lmgd0f-0 giZWgM Box-sc-6sssmg-0 inpIDs"
        display="inline-block"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <div
          className="Box-sc-6sssmg-0 inpIDs"
        >
          <svg
            className="sc-bdVaJa fJZkWb"
            viewBox="0 0 36 36"
          >
            <path
              className="sc-bwzfXH fUxNdF"
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <path
              className="sc-bxivhb MebOY"
              color={[Function]}
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <text
              className="sc-htpNat jJPiwm"
              x="18"
              y="20.35"
            >
              mockPP
            </text>
          </svg>
          <div
            className="Box-sc-6sssmg-0 inpIDs"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            PP
          </div>
        </div>
      </div>
      <div
        className="styled__Column-sc-1lmgd0f-0 giZWgM Box-sc-6sssmg-0 inpIDs"
        display="inline-block"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <div
          className="Box-sc-6sssmg-0 inpIDs"
        >
          <svg
            className="sc-bdVaJa fJZkWb"
            viewBox="0 0 36 36"
          >
            <path
              className="sc-bwzfXH fUxNdF"
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <path
              className="sc-bxivhb umeSt"
              color={[Function]}
              d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
            />
            <text
              className="sc-htpNat jJPiwm"
              x="18"
              y="20.35"
            >
              mockPower
            </text>
          </svg>
          <div
            className="Box-sc-6sssmg-0 inpIDs"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Power
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
